File: Proyecto LP/mi-servidor/Frontend/lib/Enums/MarcaDeAuto.dart
enum MarcaDeAutoEnum {
  audi("Audi", ["A1", "A3", "A4", "Q3", "Q5", "Q7"]),
  bmw("BMW", ["Series 1", "Series 3", "Series 5", "X1", "X3", "X5"]),
  chevrolet("Chevrolet", ["Spark", "Aveo", "Cruze", "Sail", "Captiva", "Tracker"]),
  ferrari("Ferrari", ["488", "F8", "Roma", "Portofino", "SF90"]),
  ford("Ford", ["F-150", "Ranger", "Explorer", "Escape", "Mustang"]),
  honda("Honda", ["Civic", "CR-V", "HR-V", "Accord", "Pilot"]),
  hyundai("Hyundai", ["Elantra", "Tucson", "Santa Fe", "Kona", "Accent"]),
  jeep("Jeep", ["Wrangler", "Cherokee", "Grand Cherokee", "Compass", "Renegade"]),
  kia("Kia", ["Rio", "Sportage", "Sorento", "Seltos", "Picanto"]),
  maserati("Maserati", ["Ghibli", "Levante", "Quattroporte", "GranTurismo", "GranCabrio"]),
  mazda("Mazda", ["Mazda2", "Mazda3", "Mazda6", "CX-3", "CX-5", "CX-9"]),
  nissan("Nissan", ["Versa", "Sentra", "Altima", "Maxima", "Pathfinder", "Frontier"]),
  peugeot("Peugeot", ["208", "308", "3008", "5008"]),
  renault("Renault", ["Kwid", "Duster", "Captur", "Logan", "Sandero"]),
  soueast("Soueast", ["DX3", "DX7", "V5"]),
  toyota("Toyota", ["Corolla", "Hilux", "Fortuner", "Yaris", "Prado"]),
  volkswagen("Volkswagen", ["Polo", "Golf", "Passat", "Tiguan", "Touareg"]),
  volvo("Volvo", ["S60", "S90", "XC40", "XC60", "XC90"]);

  final String displayName;
  final List<String> models;

  const MarcaDeAutoEnum(this.displayName, this.models);

  // Método para obtener el enum desde el displayName
  static MarcaDeAutoEnum? fromDisplayName(String? displayName) {
    return MarcaDeAutoEnum.values.firstWhere(
      (marca) => marca.displayName == displayName,
      orElse: () => throw Exception("Marca no válida: $displayName"),
    );
  }


  
}

File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/MarcaDeAuto.java
package com.miapp.mi_servidor.Enums;

import java.util.ArrayList;

public enum MarcaDeAuto {
    AUDI("Audi", new ArrayList<String>() {
        {
            addLast("A1");
            addLast("A3");
            addLast("A4");
            addLast("Q3");
            addLast("Q5");
            addLast("Q7");
        }
    }),
    BMW("BMW", new ArrayList<String>() {
        {
            addLast("Series 1");
            addLast("Series 3");
            addLast("Series 5");
            addLast("X1");
            addLast("X3");
            addLast("X5");
        }
    }),
    CHEVROLET("Chevrolet", new ArrayList<String>() {
        {
            addLast("Spark");
            addLast("Aveo");
            addLast("Cruze");
            addLast("Sail");
            addLast("Captiva");
            addLast("Tracker");
        }
    }),
    FERRARI("Ferrari", new ArrayList<String>() {
        {
            addLast("488");
            addLast("F8");
            addLast("Roma");
            addLast("Portofino");
            addLast("SF90");
        }
    }),
    FORD("Ford", new ArrayList<String>() {
        {
            addLast("F-150");
            addLast("Ranger");
            addLast("Explorer");
            addLast("Escape");
            addLast("Mustang");
        }
    }),
    HONDA("Honda", new ArrayList<String>() {
        {
            addLast("Civic");
            addLast("CR-V");
            addLast("HR-V");
            addLast("Accord");
            addLast("Pilot");
        }
    }),
    HYUNDAI("Hyundai", new ArrayList<String>() {
        {
            addLast("Elantra");
            addLast("Tucson");
            addLast("Santa Fe");
            addLast("Kona");
            addLast("Accent");
        }
    }),
    JEEP("Jeep", new ArrayList<String>() {
        {
            addLast("Wrangler");
            addLast("Cherokee");
            addLast("Grand Cherokee");
            addLast("Compass");
            addLast("Renegade");
        }
    }),
    KIA("Kia", new ArrayList<String>() {
        {
            addLast("Rio");
            addLast("Sportage");
            addLast("Sorento");
            addLast("Seltos");
            addLast("Picanto");
        }
    }),
    MASERATI("Maserati", new ArrayList<String>() {
        {
            addLast("Ghibli");
            addLast("Levante");
            addLast("Quattroporte");
            addLast("GranTurismo");
            addLast("GranCabrio");
        }
    }),
    MAZDA("Mazda", new ArrayList<String>() {
        {
            addLast("Mazda2");
            addLast("Mazda3");
            addLast("Mazda6");
            addLast("CX-3");
            addLast("CX-5");
            addLast("CX-9");
        }
    }),
    NISSAN("Nissan", new ArrayList<String>() {
        {
            addLast("Versa");
            addLast("Sentra");
            addLast("Altima");
            addLast("Maxima");
            addLast("Pathfinder");
            addLast("Frontier");
        }
    }),
    PEUGEOT("Peugeot", new ArrayList<String>() {
        {
            addLast("208");
            addLast("308");
            addLast("3008");
            addLast("5008");
        }
    }),
    RENAULT("Renault", new ArrayList<String>() {
        {
            addLast("Kwid");
            addLast("Duster");
            addLast("Captur");
            addLast("Logan");
            addLast("Sandero");
        }
    }),
    SOUEAST("Soueast", new ArrayList<String>() {
        {
            addLast("DX3");
            addLast("DX7");
            addLast("V5");
        }
    }),
    TOYOTA("Toyota", new ArrayList<String>() {
        {
            addLast("Corolla");
            addLast("Hilux");
            addLast("Fortuner");
            addLast("Yaris");
            addLast("Prado");
        }
    }),
    VOLKSWAGEN("Volkswagen", new ArrayList<String>() {
        {
            addLast("Polo");
            addLast("Golf");
            addLast("Passat");
            addLast("Tiguan");
            addLast("Touareg");
        }
    }),
    VOLVO("Volvo", new ArrayList<String>() {
        {
            addLast("S60");
            addLast("S90");
            addLast("XC40");
            addLast("XC60");
            addLast("XC90");
        }
    });

    private String name;
    private ArrayList<String> models;

    MarcaDeAuto(String name, ArrayList<String> models) {
        this.name = name;
        this.models = models;
    }

    public String getName() {
        return name;
    }

    public ArrayList<String> getModels() {
        return models;
    }

    @Override
    public String toString() {
        return name + " models: " + models.toString();
    }

    public static MarcaDeAuto fromValue(String value) {
        for (MarcaDeAuto marca : MarcaDeAuto.values()) {
            if (marca.getName().equalsIgnoreCase(value)) {
                return marca;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum MarcaDeAuto: " + value);
    }

}